src/Workspaces/Core/Portable/Options/DocumentOptionSet.cs

```diff
@@ -15,28 +20,79 @@ namespace Microsoft.CodeAnalysis.Options
     /// </summary>
     public sealed class DocumentOptionSet : OptionSet
     {
-        private readonly OptionSet _backingOptionSet;
```

Comment:
📝 I didn't fully verify the correctness of inlining this as `DocumentSpecificOptionSet`